File: employee_app/Expense.py

class Expense:
    id_counter = 0
    def __init__(self, employee_id, amount, description, date):
        self.employee_id = employee_id
        self.amount = amount
        self.description = description
        self.date = date
        self.id = self.id_counter
        self.id_counter += 1 
    
    def __str__(self):
        return f"Expsense: {self.id}, {self.description} for + ${self.amount:.2f} on {self.date}"

File: manager_app/manager_app/src/main/java/com/rev/dao/model/Expense.java
package com.rev.dao.model;

public class Expense {
    private final int id;
    private final int userId;
    private final double amount;
    private final String category;
    private final String description;
    private final String date;

    public Expense(int id, int userId, double amount, String category,  String description, String date) {
        this.id = id;
        this.userId = userId;
        this.amount = amount;
        this.category = category;
        this.description = description;
        this.date = date;
    }

    public int getId() {
        return id;
    }

    public int getUserId() {
        return userId;
    }

    public double getAmount() {
        return amount;
    }

    public String getCategory() {
        return category;
    }

    public String getDescription() {
        return description;
    }

    public String getDate() {
        return date;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Expense{");
        sb.append("id=").append(id);
        sb.append(", userId=").append(userId);
        sb.append(", amount=").append(amount);
        sb.append(", category=").append(category);
        sb.append(", description=").append(description);
        sb.append(", date=").append(date);
        sb.append('}');
        return sb.toString();
    }



}
